### PR TITLE
✨Implement Error Handling For Screen

### LINKED ISSUE #328: Fix return values for Screen API
#### Expected Behavior:
Screen API does not set errno, or return PROS_ERR in any case. It just returns the raw value from the VEX syscalls.

### PR DESCRIPTION
**Pull request by AurinDutta**
#### Summary:
Implemented error catching and handling for the screen class, where if an error is detected (e.g. a mutex isn't able to be taken), the method in question will catch the error, set errno to the proper error, and return PROS_ERR. 

#### Motivation:
The screen class didn't have any error catching or handling, as such this addition will help users when debugging their code.

##### References (optional):
closes #328 

#### Test Plan:
While a mutex is taken, run methods from the screen class; if they set errno correctly and return PROS_ERR, then the error handling works fine.

- [ ] test item

### CODE REVIEW

File: include/pros/screen.h [hunk 1]
```diff
@@ -96,46 +97,82 @@ namespace c {
...
  * Set the pen color for subsequent graphics operations
+ * 
+ * This function uses the following values of errno when an error state is
+ * reached:
+ * EACCESS - Another resource is currently trying to access the screen mutex.
  *
  * \param color	The pen color to set (it is recommended to use values
  * 		 from the enum defined in colors.h)
+ * 
+ * \return Returns 1 if the mutex was successfully returned, or prosERR if there was an error either taking or
```
Comment (HotelCalifornia) [inline include/pros/screen.h:108]: PROS_ERR not prosERR (here and throughout the file)

File: include/pros/screen.h [hunk 2]
```diff
@@ -144,107 +181,179 @@ void screen_scroll(int16_t start_line, int16_t lines);
...
+ * reached:
+ * EACCESS - Another resource is currently trying to access the screen mutex.
+ *
  * \param x0, y0	The (x,y) coordinates of the first corner of the
  * 						rectangular region
  * \param x1, y1	The (x,y) coordinates of the second corner of the
  * 						rectangular region
  * \param lines 	The number of lines to scroll upwards
+ * 
+ * \return 1 if there were no errors, or PROS_ERR if an error occured taking or returning the screen mutex.
```
Comment (HotelCalifornia) [inline include/pros/screen.h:194]: Make sure your comments aren't going past column 80 on any line

**Review by HotelCalifornia — COMMENTED** [review]
some nitpicks with doc comments

File: include/pros/screen.h [hunk 3]
```diff
@@ -46,7 +46,8 @@ typedef enum {
 typedef enum {
     E_TOUCH_RELEASED = 0, ///< Last interaction with screen was a quick press
     E_TOUCH_PRESSED, ///< Last interaction with screen was a release
-    E_TOUCH_HELD ///< User is holding screen down 
+    E_TOUCH_HELD, ///< User is holding screen down
+    MUTEX_ERROR // An error occured while taking/returning the mutex
```
[the remark this answers is not in the document]
Comment (WillXuCodes) [reply include/pros/screen.h:50]: Also change it to E_TOUCH_ERROR or something more in line with the other enum options.

File: src/devices/screen.c [hunk 4]
```diff
@@ -293,10 +456,13 @@ static inline bool _touch_status_equivalent(V5_TouchStatus x, V5_TouchStatus y)
 	return (x.lastEvent == y.lastEvent) && (x.lastXpos == y.lastXpos) && (x.lastYpos == y.lastYpos);
 }
 
-void _touch_handle_task(void* ignore) {
+uint32_t _touch_handle_task(void* ignore) {
```
Comment (WillXuCodes) [inline src/devices/screen.c:459]: Given that this is a daemon for handling touch callbacks, I don't think we really need this to be changed to have a return value.

**Review by WillXuCodes — CHANGES_REQUESTED** [review]

**Review by WillXuCodes — DISMISSED** [review]
LGTM

**Review by WillXuCodes — APPROVED** [review]
Thanks for fixing the comment, looks good.

**Pull request merged**